src/pages/Application/ApplicationForm.js

```diff
@@ -26,65 +27,99 @@ class ApplicationForm extends Component {
+                    </div>
+                </div>
+                <div className="app-row">
+                    <div className="col-6 paddingr">
+                      <label>Upload Resume</label>
+                      <button type="button" onClick={() => { dropzoneRef.open() }} className="application-button">Drop or click to upload</button>
+                      <ResumeUploadProgressIndicator/>
+                      { applicationForm.resume_uploaded ? <div>You've uploaded <a href={applicationForm.resume_get_url} target="_blank" rel="noopener noreferrer" >{applicationForm.resume_filename}</a></div> : null }
+                    </div>
+                    <div className="col-6">
+                      <label>First Hackathon?</label>
+                      <ApplicationTextField field="is_first_hackathon"/>
+                    </div>
+                </div>
+                <div className="app-row">
+                    <div className="col-6 paddingr">
+                      <label>LinkedIn Username</label>
+                      { applicationForm.has_no_linkedin ?
+                          <div>You indicated you don't have a Linedin <button onClick={this.toggleItem.bind(this,'has_no_linkedin')} className="opt-in-button">I do!</button></div>
+                      :
+                          <div>
+                              <ApplicationTextField field="linkedin" styles={ { 'border-bottom-left-radius': 0, 'border-bottom-right-radius': 0 } }/>
+                              <button onClick={this.toggleItem.bind(this,'has_no_linkedin')} className="opt-out-button">No thanks</button>
+                          </div>
+                      }
+                    </div>
+                    <div className="col-6">
+                      <label>GitHub Username</label>
+                      { applicationForm.has_no_github && !isGithubLinked ?
+                          <div>You indicated you don't have a Github <button onClick={this.toggleItem.bind(this,'has_no_github')} className="opt-in-button">I do!</button></div>
+                      :
+                          <div>
+                              <ApplicationTextField field="github" disabled={isGithubLinked} styles={ { 'border-bottom-left-radius': 0, 'border-bottom-right-radius': 0 } }/>
+                              { isGithubLinked ?
+                                  <i>You signed up with github, so you can't change the username</i>
+                              :
+                                  <button onClick={this.toggleItem.bind(this,'has_no_github')} className="opt-out-button">No thanks</button>
+                              }
+                          </div>
+                      }
+                    </div>
+                </div>
+                <div className="app-row">
+                    <div className="col-6">
+                        <label>Gender</label>
+                        {/* <Select ref="schoolSelect"
```

Comment:
i know it's commented out right now, but this should end up in it's own component, even though it isn't reusable, it's cleaner and keeps redux logic outside of `<ApplicationForm/>`